pkgs/tools/video/yamdi/default.nix

```diff
@@ -0,0 +1,36 @@
+{
+  stdenv,
+  fetchurl,
+  ...
+}:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "yamdi-${version}";
+  version = "1.9";
+
+  # Source repo is also available here:
+  # https://github.com/ioppermann/yamdi
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/project/yamdi/yamdi/${version}/yamdi-${version}.tar.gz";
+    sha256 = "4a6630f27f6c22bcd95982bf3357747d19f40bd98297a569e9c77468b756f715";
+  };
+
+  buildFlags = "CC=${stdenv.cc}/bin/cc";
```

Comment:
Without this, my build fails (on Darwin, but that should be irrelevant). I
can add `which` to buildInputs and put the following in my installPhase:

```
echo cc: `which cc`
echo gcc: `which gcc`
```

And during the build it prints this:

```
cc: /nix/store/c06lj2kr7s31wav5m958qr9kkl0cd73a-clang-wrapper-3.7.1/bin/cc
which: no gcc in
(/nix/store/gf8qhgnr26q8xi4zyshcz04fn3qba9vl-which-2.21/bin:/nix/store/c06lj2kr7s31wav5m958qr9kkl0cd73a-clang-wrapper-3.7.1/bin:/nix/store/clqq5iva8aylmr3k08lpp68ai7g6wz2r-coreutils-8.25/bin:/nix/store/kjpflxd2229qg37cfqzdijbgmq1xh45g-findutils-4.4.2/bin:/nix/store/dzv0k8dmd6icm98x2xb1pl9nf1gsdcgi-diffutils-3.3/bin:/nix/store/max5llaf3c9f7v5mb6g91c35a8434qb5-gnused-4.2.2/bin:/nix/store/f2dpg02w3j7hb4wj2knjpii12khc89z4-gnugrep-2.22/bin:/nix/store/i0q5wky82v16b4x47jm2ak020vq9bcyi-gawk-4.1.3/bin:/nix/store/ks30pk1mhp79lh10i1dsjvi54nv4c98x-gnutar-1.28/bin:/nix/store/gjniklqbnwhqckfmvy1gws3d0nrabwc7-gzip-1.6/bin:/nix/store/yavc7lgyys2xdjqy9ns55jswn7rr3sqy-bzip2-1.0.6/bin:/nix/store/gn3skk7cvpdafsf557ppi4iqpz4dpx1s-gnumake-4.1/bin:/nix/store/dv3f65rk4adnx85m0hhrbqcy3f1gbc2x-bash-4.3-p42/bin:/nix/store/r9q7mw8d7ird19j74snbks8s1p41rcyc-patch-2.7.5/bin:/nix/store/dfcmc4ic4xxlc83dr2cnjf2p911wmyx3-xz-5.2.2/bin:/nix/store/sby0adgbn80wixajvgbgj8k06j0z7b87-clang-3.7.1/bin:/nix/store/n1f9dm9210nd29rpzhab5s426i7vcn5n-cctools-binutils-darwin/bin:/nix/store/clqq5iva8aylmr3k08lpp68ai7g6wz2r-coreutils-8.25/bin)
gcc:
```

So it looks like I need to somehow override the `CC=gcc` in the Makefile,
but I suppose I could simplify to `buildFlags = "CC=cc";`. Just tested, and
that works equally well.


On Wed, Mar 16, 2016 at 11:05 AM, Christoph Hrdinka <
notifications@github.com> wrote:

> In pkgs/tools/video/yamdi/default.nix
> <https://github.com/NixOS/nixpkgs/pull/13962#discussion_r56350010>:
>
> > +}:
> > +
> > +with stdenv.lib;
> > +
> > +stdenv.mkDerivation rec {
> > +  name = "yamdi-${version}";
> > +  version = "1.9";
> > +
> > +  # Source repo is also available here:
> > +  # https://github.com/ioppermann/yamdi
> > +  src = fetchurl {
> > +    url = "https://downloads.sourceforge.net/project/yamdi/yamdi/${version}/yamdi-${version}.tar.gz";
> > +    sha256 = "4a6630f27f6c22bcd95982bf3357747d19f40bd98297a569e9c77468b756f715";
> > +  };
> > +
> > +  buildFlags = "CC=${stdenv.cc}/bin/cc";
>
> Is this really needed? yamdi's Makefile sets this to gcc, there shouldn't
> be a problem with that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/13962/files/40a1d6e14554c0b13631738b3c696f00a8f2123d#r56350010>
>



-- 
Ryan Artecona